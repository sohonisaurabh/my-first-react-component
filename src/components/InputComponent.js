import React from "react";

class InputComponent extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            val: "Default Value"
        };
        this.handleChange = this.handleChange.bind(this);
    }
    handleChange(event) {
        this.setState({
            val: event.target.value
        });
    }
    render () {
        return (
            <input type="text" value={this.state.val} onChange={this.handleChange} />
        );
    }
}

/* const InputComponent = (props) => {
    return (
        <input type="text" value="{props.defaultVal}" onChange={props.handleChange}/>
    );
}; */

export default InputComponent;