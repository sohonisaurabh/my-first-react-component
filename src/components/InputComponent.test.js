import React from "react";
import {shallow} from "enzyme";
import InputComponent from "./InputComponent";

describe("InputComponentTestSuite", () => {
    it("should render input text component", () => {
        const comp = shallow(<InputComponent />);
        expect(comp.getElements()).toMatchSnapshot();
    });
});