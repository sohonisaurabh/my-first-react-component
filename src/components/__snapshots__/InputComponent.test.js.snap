// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponentTestSuite should render input text component 1`] = `
Array [
  <input
    onChange={[Function]}
    type="text"
    value="Default Value"
  />,
]
`;
